Mathlib/Geometry/Manifold/IsManifold/Basic.lean

```diff
@@ -822,8 +822,11 @@ variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M']
 
 /-- The disjoint union of two `C^n` manifolds modelled on `(E, H)`
 is a `C^n` manifold modeled on `(E, H)`. -/
-instance disjointUnion [Nonempty H] : IsManifold I n (M ⊕ M') where
+instance disjointUnion : IsManifold I n (M ⊕ M') where
   compatible {e} e' he he' := by
+    by_cases h : Nonempty H; swap
```

Comment:
Isn't there a `isEmpty_or_nonempty` to make the disjunction more direct?

Reply:
Indeed! Changed.